x-pack/plugins/fleet/server/services/agents/update_agent_tags_action_runner.ts

```diff
@@ -57,70 +83,125 @@ export async function updateTagsBatch(
     tagsToRemove: string[];
     actionId?: string;
     total?: number;
+    kuery?: string;
   }
-): Promise<{ actionId: string }> {
+): Promise<{ actionId: string; updated?: number; took?: number }> {
   const errors: Record<Agent['id'], Error> = { ...outgoingErrors };
+  const hostedAgentError = `Cannot modify tags on a hosted agent`;
 
   const filteredAgents = await filterHostedPolicies(
     soClient,
     givenAgents,
     errors,
-    `Cannot modify tags on a hosted agent`
+    hostedAgentError
   );
+  const agentIds = filteredAgents.map((agent) => agent.id);
+
+  let query: estypes.QueryDslQueryContainer | undefined;
+  if (options.kuery !== undefined) {
+    const hostedPolicies = await agentPolicyService.list(soClient, {
```

Comment:
Looks like by default the `agentPolicyService.list` return only the 20 first policies,  we may want to pass a higher limit here

Reply:
good catch, I'll update